File: ios/MindfulMoves/WidgetDataModule.m
//
//  WidgetDataModule.m
//  MindfulMoves
//
//  Objective-C bridge for WidgetDataModule
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(WidgetDataModule, NSObject)

RCT_EXTERN_METHOD(writeWidgetData:(NSString *)data
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/MindfulMoves/WidgetDataModule.swift
//
//  WidgetDataModule.swift
//  MindfulMoves
//
//  Native module to write widget data to App Group UserDefaults
//  This allows the widget extension to read the data
//

import Foundation
import React

@objc(WidgetDataModule)
class WidgetDataModule: NSObject {
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
  @objc
  func writeWidgetData(_ data: String, resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {
    // Write to App Group UserDefaults (shared with widget extension)
    if let sharedDefaults = UserDefaults(suiteName: "group.com.mindfulmoves.app") {
      // Convert JSON string to Data
      if let jsonData = data.data(using: .utf8) {
        sharedDefaults.set(jsonData, forKey: "widget_habits")
        sharedDefaults.synchronize()
        resolver(true)
      } else {
        rejecter("ENCODING_ERROR", "Failed to encode widget data", nil)
      }
    } else {
      rejecter("APP_GROUP_ERROR", "App Group not configured", nil)
    }
  }
}

